### PR TITLE
Add `ParetoNBDModel`

### PR DESCRIPTION
**Pull request by ColtAllen**
Reopening this PR for the third time because I got my local working branches confused.

Still resolving the testing changes; for some reason an instantiated model returns a `NoneType`. Once that's resolved it should be smooth sailing.

I've also created an updated UML diagram reflecting the `ModelBuilder` integrations in `BaseModel`:

classes_clv

----
:books: Documentation preview :books:: https://pymc-marketing--312.org.readthedocs.build/en/312/

### CODE REVIEW

**Comment by ColtAllen** [comment]
All tests are now passing! Turns out this usage convention is required:

```
model = ParetoNBDModel(data)
model.build_model()
```

Because `model = ParetoNBDModel(data).build_model()` will return a `NoneType`. I'll consider revising this in a future PR.

Also, test coverage is 100% if the `runslow` tests are included, but 89% otherwise. This may require some CI changes per https://github.com/pymc-labs/pymc-marketing/issues/308.

File: pymc_marketing/clv/models/pareto_nbd.py [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+            rfm_df = rfm_summary(raw_data,'id_col_name','date_col_name')
+
+            model = ParetoNBDModel(
+                data=rfm_df,
+                model_config={
+                r=pm.Weibull.dist(alpha=2,beta=1),
+                alpha=pm.Weibull.dist(alpha=2,beta=10),
+                s=pm.Weibull.dist(alpha=2,beta=1),
+                beta=pm.Weibull.dist(alpha=2,beta=10),
+                },
```
Comment (ricardoV94) [inline pymc_marketing/clv/models/pareto_nbd.py:105]: The example is outdated (possibly others in the docstrings as well)?

File: pymc_marketing/clv/models/pareto_nbd.py [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+                sampler_config={
+                    "draws": 1000,
+                    "tune": 1000,
+                    "chains": 2,
+                    "cores": 2,
+                    "nuts_kwargs": {"target_accept": 0.95},
+                },
```
Comment (ricardoV94) [inline pymc_marketing/clv/models/pareto_nbd.py:112]: I don't think this is needed in the example. Less can be more!

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Only some old docstrings that need to be updated. Functionality wise looks kosher!

File: pymc_marketing/clv/models/pareto_nbd.py (hunk 1 above)
Comment (ColtAllen) [reply pymc_marketing/clv/models/pareto_nbd.py:105]: Parameter names are up to date, but that dict argument for `model_config` isn't formatted correctly. I'll go ahead and fix it.

File: pymc_marketing/clv/models/pareto_nbd.py (hunk 2 above)
Comment (ColtAllen) [reply pymc_marketing/clv/models/pareto_nbd.py:112]: Sampler configurations can make or break the `ParetoNBDModel` as we've discovered in the dev notebook. However, the slow tests converge fine with the default configs, so I'll remove this for now and add recommendations in a future PR if issues arise.

File: pymc_marketing/clv/models/pareto_nbd.py [hunk 3]
```diff
@@ -98,17 +98,10 @@ class ParetoNBDModel(CLVModel):
...
-                sampler_config={
-                    "draws": 1000,
-                    "tune": 1000,
-                    "chains": 2,
-                    "cores": 2,
-                    "nuts_kwargs": {"target_accept": 0.95},
+                "r": pm.Weibull.dist(alpha=2,beta=1),
+                "alpha": pm.Weibull.dist(alpha=2,beta=10),
+                "s": pm.Weibull.dist(alpha=2,beta=1),
+                "beta": pm.Weibull.dist(alpha=2,beta=10),
```
Comment (ricardoV94) [inline pymc_marketing/clv/models/pareto_nbd.py:104]:
```suggestion
                    "r": pm.Weibull.dist(alpha=2,beta=1),
                    "alpha": pm.Weibull.dist(alpha=2,beta=10),
                    "s": pm.Weibull.dist(alpha=2,beta=1),
                    "beta": pm.Weibull.dist(alpha=2,beta=10),
```

File: pymc_marketing/clv/models/pareto_nbd.py (hunk 1 above)
Comment (ricardoV94) [reply pymc_marketing/clv/models/pareto_nbd.py]: I didn't even notice it was inside a dict because of the lack of identation

**Comment by ColtAllen** [comment]
I noticed the predictive methods were still using the old individual column convention for inputs. Just updated them along with the `_process_customers` utility for dataframe inputs.

**Pull request merged**